docs/lifecycle.md

```diff
@@ -0,0 +1,278 @@
+# Object Lifecycle Management
+
+## OVERVIEW
+
+The purpose is to apply various lifecycle actions to objects after a
+certain time has passed since object creation, per lifecycle rules as
+specified in AWS S3 APIs.
+
+### EXPIRATION
+
+We support expiration of versioned or non-versioned objects, when a
+number of days have passed since their creation.
+
+One important use case is to allow automatic deletion of older
+versions of versioned objects to reclaim storage space.
```

Comment:
Will there be an option for versions expiration (without deleting the object itself)? For example, one updates a versioned object every day, and wants to get rid of versions older than three months? I believe there's a field usecase for that, but I'm unsure whether this has been discussed or not (it might simply be possible by applying the policy to each version with its id?)

Reply:
What you describe is how it's going to work, actually, and it's the first use case we implement lifecycle for, to delete older versions that CRR has created on the backup site. Maybe the description is confusing, let me know if you think of a better formulation.

The API is going to support what AWS supports for expiration: https://docs.aws.amazon.com/AmazonS3/latest/dev/intro-lifecycle-rules.html

The implementation will be based (for now at least) on a cron-style task that wakes up periodically (say one or more times per day), lists buckets to get their objects and their versions, match against bucket lifecycle configurations, and take appropriate actions when there is a match.